cpp/daal/src/algorithms/svm/svm_train_thunder_cache.h

```diff
@@ -184,16 +203,18 @@ class SVMCache<thunder, lruCache, algorithmFPType, cpu> : public SVMCacheIface<t
         _kernelOriginalIndex.reset(nSize);
         DAAL_CHECK_MALLOC(_kernelOriginalIndex.get());
 
-        _cacheData.reset(_lineSize * _cacheSize);
+        const size_t maxAlignedSize = 64 >> getDegreeAlgorithmFPType();
+        _cacheData.reset((_lineSize + maxAlignedSize) * _cacheSize);
         DAAL_CHECK_MALLOC(_cacheData.get());
 
-        _cache = SOANumericTableCPU<cpu>::create(_cacheSize, _lineSize, DictionaryIface::FeaturesEqual::equal, &status);
-        DAAL_CHECK_STATUS_VAR(status);
+        _cache.reset(_cacheSize);
+        DAAL_CHECK_MALLOC(_cache.get());
 
         for (size_t i = 0; i < _cacheSize; ++i)
         {
-            auto cachei = &_cacheData[i * _lineSize];
-            DAAL_CHECK_STATUS(status, _cache->template setArray<algorithmFPType>(cachei, i));
+            algorithmFPType * const cachei = &_cacheData[i * (maxAlignedSize + _lineSize)];
+            const size_t align             = ((64 - (reinterpret_cast<size_t>(cachei) & 63)) & 63) >> getDegreeAlgorithmFPType();
+            _cache[i]                      = cachei + align;
```

Comment:
I can't understand the algorithm, you can try this version:

```cpp
size_t bytes = _lineSize * sizeof(algorithmFPType );                     // to bytes
size_t alignedBytesSize = bytes  & 63 ? (bytes   & (~63)) + 64 : bytes;  // nearest number aligned on 64
size_t newLineSize = alignedBytesSize  / sizeof(algorithmFPType;         // to elements
_cacheData.reset(new_line_size  * _cacheSize); 

... 
for (size_t i = 0; i < _cacheSize; ++i) {
    _cache[i]  = &_cacheData[i * newLineSize ]; // _cache[i] - always aligned on 64 bytes
}
```

Reply:
I liked your solution and I applied it.